### PR TITLE
Ibc go v7

### PR DESCRIPTION
**Pull request by dongrie**
* upgrade ibc-go to v7
* change /gogo/protobuf to /cosmos/gogoproto
* remove third_party/proto
* update proto-gen

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
+DOCKER := $(shell which docker)
+
+protoVer=0.13.1
+protoImageName=ghcr.io/cosmos/proto-builder:$(protoVer)
+protoImage=$(DOCKER) run --user 0 --rm -v $(CURDIR):/workspace --workdir /workspace $(protoImageName)
 
 .PHONY: proto-gen
 proto-gen:
 	@echo "Generating Protobuf files"
-	@mkdir -p ./proto/ibc/lightclients/ethereum/v1
```
Comment (bluele) [inline Makefile:6]: Any reason to remove these?
The proto definitions of light client need to be copied from `ethereum-ibc-rs`.

File: proto/relay/ethereum_light_client/config/config.proto [hunk 2]
```diff
@@ -1,5 +1,5 @@
 syntax = "proto3";
-package relayer.provers.ethereum_light_client.config;
+package relay.ethereum_light_client.config;
```
Comment (siburu) [inline proto/relay/ethereum_light_client/config/config.proto]: Keep the package name as original and move this file to `relay/provers/ethereum_light_client/config` directory

File: relay/prover.go [hunk 3]
```diff
@@ -324,122 +321,17 @@ func (pr *Prover) buildNextSyncCommitteeUpdateForNext(period uint64, trustedHeig
...
-	), proofHeight)
+	height := pr.newHeight(proofHeight)
+	proof, err := pr.buildStateProof([]byte(path), proofHeight)
 	if err != nil {
-		return nil, err
+		return nil, height, err
 	}
-	res.ProofHeight = pr.newHeight(proofHeight)
-	return res, nil
+	return proof, height, nil
```
Comment (siburu) [inline relay/prover.go]: You can simply rewrite this part to one line `return proof, height, err`

File: scripts/protocgen.sh [hunk 4]
```diff
@@ -2,23 +2,16 @@
...
-  --grpc-gateway_out=logtostderr=true:. \
-  $(find "${dir}" -maxdepth 1 -name '*.proto')
+buf generate --template buf.gen.gogo.yaml
 
-done
+cd ..
 
+# move proto files to the right places
 cp -r github.com/datachainlab/ethereum-ibc-relay-prover/* ./
+cp -r github.com/datachainlab/ethereum-lcp/* ./
```
Comment (siburu) [inline scripts/protocgen.sh]: This addition is not needed if the proto of light client definition is copied from ethereum-ibc-rs.

File: Makefile (hunk 1 above)
Comment (siburu) [reply Makefile:6]: @dongrie I think that if you follow this advice you can avoid adding `cp -r github.com/datachainlab/ethereum-lcp/* ./` in `scripts/protocgen.sh`.

Comment (dongrie) [reply Makefile:6]: Revert mkdir code, and remove cp command from protocgen.sh
https://github.com/datachainlab/ethereum-ibc-relay-prover/pull/3/commits/7484802a5ca3f65de988f15af8808db21dce92be

File: proto/relay/ethereum_light_client/config/config.proto (hunk 2 above)
Comment (dongrie) [reply proto/relay/ethereum_light_client/config/config.proto]: Revert package name 
https://github.com/datachainlab/ethereum-ibc-relay-prover/pull/3/commits/7484802a5ca3f65de988f15af8808db21dce92be

File: relay/prover.go (hunk 3 above)
Comment (dongrie) [reply relay/prover.go]: Remove check err
https://github.com/datachainlab/ethereum-ibc-relay-prover/pull/3/commits/7484802a5ca3f65de988f15af8808db21dce92be

File: scripts/protocgen.sh (hunk 4 above)
Comment (dongrie) [reply scripts/protocgen.sh]: Remove cp command
https://github.com/datachainlab/ethereum-ibc-relay-prover/pull/3/commits/7484802a5ca3f65de988f15af8808db21dce92be

**Comment by dongrie** [comment]
@siburu `make proto-update-deps` added
 https://github.com/datachainlab/ethereum-ibc-relay-prover/pull/3/commits/696d9f7da5357fdf2ae3fc2237c97694a5a65483

File: Makefile (hunk 1 above)
Comment (siburu) [reply Makefile:6]: I misunderstand the point. You also need revert sed code.

Comment (siburu) [reply Makefile:6]: Ah, but maybe, you also need change `go_package` from ethereum-lcp to ethereum-ibc-relay-prover.

File: Makefile [hunk 5]
```diff
@@ -1,9 +1,15 @@
-ETHEREUM_IBC_PROTO ?= ./ethereum-ibc-rs/proto/definitions
```
Comment (siburu) [inline Makefile:1]: You also need revert this line.

Comment (dongrie) [reply Makefile:1]: Revert path
https://github.com/datachainlab/ethereum-ibc-relay-prover/pull/3/commits/4361f6b7b6141dbf088cc162572a7b833e66118f

File: Makefile (hunk 1 above)
Comment (dongrie) [reply Makefile:6]: Revert sed
https://github.com/datachainlab/ethereum-ibc-relay-prover/pull/3/commits/4361f6b7b6141dbf088cc162572a7b833e66118f

File: relay/prover.go [hunk 6]
```diff
@@ -324,122 +321,14 @@ func (pr *Prover) buildNextSyncCommitteeUpdateForNext(period uint64, trustedHeig
 
 //--------- IBCProvableQuerier implementation ---------//
```
Comment (siburu) [inline relay/prover.go]: There no longer exists `IBCProvableQuerier` ... replace this with `StateProver`

File: relay/prover.go [hunk 7]
```diff
@@ -324,122 +321,14 @@ func (pr *Prover) buildNextSyncCommitteeUpdateForNext(period uint64, trustedHeig
...
-		pr.chain.Path().ChannelID,
-		seq,
-	), proofHeight)
-	if err != nil {
-		return nil, err
-	}
-	res.ProofHeight = pr.newHeight(proofHeight)
-	return res, nil
-}
+var _ core.Prover = (*Prover)(nil)
```
Comment (siburu) [inline relay/prover.go]: You should make sure here that `*Prover` implements `core.StateProver`, not `core.Prover`.

Comment (dongrie) [reply relay/prover.go]: https://github.com/datachainlab/ethereum-ibc-relay-prover/pull/3/commits/6c45a9b4de8692979da2394fbbd124f8ff2df04f
change to `StateProver`

File: relay/prover.go (hunk 6 above)
Comment (dongrie) [reply relay/prover.go]: Fix comment
https://github.com/datachainlab/ethereum-ibc-relay-prover/pull/3/commits/6c45a9b4de8692979da2394fbbd124f8ff2df04f

**Pull request merged**